code/_onclick/item_attack.dm

```diff
@@ -21,74 +19,91 @@
 			playsound(loc, 'sound/weapons/slice.ogg', 50, 1, -1)
 			if(do_mob(user, src, 80/sharpness))
 				harvest(user)
-			return
-	I.attack(src, user)
-
-/mob/living/proc/attacked_by(obj/item/I, mob/living/user, def_zone)
-	apply_damage(I.force, I.damtype, def_zone)
-	if(I.damtype == "brute")
-		if(prob(33) && I.force)
-			var/turf/location = get_turf(src)
-			location.add_blood_floor(src)
-
-	var/message_verb = ""
-	if(I.attack_verb && I.attack_verb.len)
-		message_verb = "[pick(I.attack_verb)]"
-	else if(I.force)
-		message_verb = "attacked"
-
-	var/attack_message = "[src] has been [message_verb] with [I]."
-	if(user)
-		user.do_attack_animation(src)
-		if(user in viewers(src, null))
-			attack_message = "[user] has [message_verb] [src] with [I]!"
-	if(message_verb)
-		visible_message("<span class='danger'>[attack_message]</span>",
-		"<span class='userdanger'>[attack_message]</span>")
-
-/mob/living/simple_animal/attacked_by(var/obj/item/I, var/mob/living/user)
-	if(!I.force)
-		user.visible_message("<span class='warning'>[user] gently taps [src] with [I].</span>",\
-						"<span class='warning'>This weapon is ineffective, it does no damage!</span>")
-	else if(I.force >= force_threshold && I.damtype != STAMINA)
-		..()
-	else
-		visible_message("<span class='warning'>[I] bounces harmlessly off of [src].</span>",\
-					"<span class='warning'>[I] bounces harmlessly off of [src]!</span>")
-
+			return 1
+	return I.attack(src, user) //phil235 should we return 1 if successful special attack() so we don't call afterattack?

```

Comment:
adding return, so that we can have attack return 1 or 0 and stop afterattack() from being called. Currently unused.